{"uid":"3e25484dba133222","name":"verifyUploadNewImageNegativeTest","fullName":"com.thecatapi.api.systemtests.images.uploaded.upload.UploadNewImageNegativeTests.verifyUploadNewImageNegativeTest","historyId":"34025b3ca94f1979bb9863f98c00edba","time":{"start":1754496812615,"stop":1754496815752,"duration":3137},"description":"Verify that user can not upload image with incorrect breedIds","descriptionHtml":"<p>Verify that user can not upload image with incorrect breedIds</p>\n","status":"failed","statusMessage":"Upload image request returned inappropriate status code\nExpecting 400 status code but was 201!","statusTrace":"com.thecatapi.api.exceptions.HttpException: Upload image request returned inappropriate status code\nExpecting 400 status code but was 201!\n\tat com.thecatapi.api.core.restassuredclient.HttpResponseWrapper.validateResponseCode(HttpResponseWrapper.java:101)\n\tat com.thecatapi.api.core.restassuredclient.HttpResponseWrapper.expectStatusCode(HttpResponseWrapper.java:17)\n\tat com.thecatapi.api.core.thecatapi.controllers.ImageController.uploadImage(ImageController.java:31)\n\tat com.thecatapi.api.systemtests.images.uploaded.upload.UploadNewImageNegativeTests.verifyUploadNewImageNegativeTest(UploadNewImageNegativeTests.java:27)\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\n\tat org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)\n\tat org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)\n\tat org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)\n\tat org.testng.internal.invokers.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:81)\n\tat org.testng.internal.thread.Async$AsyncTask.run(Async.java:34)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)\n\tat java.base/java.lang.Thread.run(Thread.java:1447)\nCaused by: java.lang.AssertionError: 1 expectation failed.\nExpected status code <400> but was <201>.\n\n\tat java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)\n\tat java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)\n\tat java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)\n\tat org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)\n\tat org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:108)\n\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)\n\tat io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure.validate(ResponseSpecificationImpl.groovy:512)\n\tat io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure$validate$5.call(Unknown Source)\n\tat io.restassured.internal.ResponseSpecificationImpl.validateResponseIfRequired(ResponseSpecificationImpl.groovy:696)\n\tat io.restassured.internal.ResponseSpecificationImpl.this$2$validateResponseIfRequired(ResponseSpecificationImpl.groovy)\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\n\tat org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)\n\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:198)\n\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:62)\n\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)\n\tat io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:135)\n\tat io.restassured.internal.ValidatableResponseOptionsImpl.statusCode(ValidatableResponseOptionsImpl.java:84)\n\tat com.thecatapi.api.core.restassuredclient.HttpResponseWrapper.validateResponseCode(HttpResponseWrapper.java:99)\n\t... 13 more\n","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[],"testStage":{"description":"Verify that user can not upload image with incorrect breedIds","status":"failed","statusMessage":"Upload image request returned inappropriate status code\nExpecting 400 status code but was 201!","statusTrace":"com.thecatapi.api.exceptions.HttpException: Upload image request returned inappropriate status code\nExpecting 400 status code but was 201!\n\tat com.thecatapi.api.core.restassuredclient.HttpResponseWrapper.validateResponseCode(HttpResponseWrapper.java:101)\n\tat com.thecatapi.api.core.restassuredclient.HttpResponseWrapper.expectStatusCode(HttpResponseWrapper.java:17)\n\tat com.thecatapi.api.core.thecatapi.controllers.ImageController.uploadImage(ImageController.java:31)\n\tat com.thecatapi.api.systemtests.images.uploaded.upload.UploadNewImageNegativeTests.verifyUploadNewImageNegativeTest(UploadNewImageNegativeTests.java:27)\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\n\tat org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)\n\tat org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)\n\tat org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)\n\tat org.testng.internal.invokers.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:81)\n\tat org.testng.internal.thread.Async$AsyncTask.run(Async.java:34)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)\n\tat java.base/java.lang.Thread.run(Thread.java:1447)\nCaused by: java.lang.AssertionError: 1 expectation failed.\nExpected status code <400> but was <201>.\n\n\tat java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)\n\tat java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)\n\tat java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)\n\tat org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)\n\tat org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:108)\n\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)\n\tat io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure.validate(ResponseSpecificationImpl.groovy:512)\n\tat io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure$validate$5.call(Unknown Source)\n\tat io.restassured.internal.ResponseSpecificationImpl.validateResponseIfRequired(ResponseSpecificationImpl.groovy:696)\n\tat io.restassured.internal.ResponseSpecificationImpl.this$2$validateResponseIfRequired(ResponseSpecificationImpl.groovy)\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\n\tat org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)\n\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:198)\n\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:62)\n\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)\n\tat io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:135)\n\tat io.restassured.internal.ValidatableResponseOptionsImpl.statusCode(ValidatableResponseOptionsImpl.java:84)\n\tat com.thecatapi.api.core.restassuredclient.HttpResponseWrapper.validateResponseCode(HttpResponseWrapper.java:99)\n\t... 13 more\n","steps":[{"name":"Send upload new image with x-api-key and with multipart as: ImageRequestDto(img=src/main/resources/img/large.jpeg, subId=null, breedIds=Ojos Azules)","time":{"start":1754496812617,"stop":1754496815752,"duration":3135},"status":"failed","statusMessage":"Upload image request returned inappropriate status code\nExpecting 400 status code but was 201!","statusTrace":"com.thecatapi.api.exceptions.HttpException: Upload image request returned inappropriate status code\nExpecting 400 status code but was 201!\n\tat com.thecatapi.api.core.restassuredclient.HttpResponseWrapper.validateResponseCode(HttpResponseWrapper.java:101)\n\tat com.thecatapi.api.core.restassuredclient.HttpResponseWrapper.expectStatusCode(HttpResponseWrapper.java:17)\n\tat com.thecatapi.api.core.thecatapi.controllers.ImageController.uploadImage(ImageController.java:31)\n\tat com.thecatapi.api.systemtests.images.uploaded.upload.UploadNewImageNegativeTests.verifyUploadNewImageNegativeTest(UploadNewImageNegativeTests.java:27)\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\n\tat org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)\n\tat org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)\n\tat org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)\n\tat org.testng.internal.invokers.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:81)\n\tat org.testng.internal.thread.Async$AsyncTask.run(Async.java:34)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)\n\tat java.base/java.lang.Thread.run(Thread.java:1447)\nCaused by: java.lang.AssertionError: 1 expectation failed.\nExpected status code <400> but was <201>.\n\n\tat java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)\n\tat java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)\n\tat java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)\n\tat org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)\n\tat org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:108)\n\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)\n\tat io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure.validate(ResponseSpecificationImpl.groovy:512)\n\tat io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure$validate$5.call(Unknown Source)\n\tat io.restassured.internal.ResponseSpecificationImpl.validateResponseIfRequired(ResponseSpecificationImpl.groovy:696)\n\tat io.restassured.internal.ResponseSpecificationImpl.this$2$validateResponseIfRequired(ResponseSpecificationImpl.groovy)\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\n\tat org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)\n\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:198)\n\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:62)\n\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)\n\tat io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:135)\n\tat io.restassured.internal.ValidatableResponseOptionsImpl.statusCode(ValidatableResponseOptionsImpl.java:84)\n\tat com.thecatapi.api.core.restassuredclient.HttpResponseWrapper.validateResponseCode(HttpResponseWrapper.java:99)\n\t... 13 more\n","steps":[],"attachments":[{"uid":"ea70788ed894760f","name":"Request Logs","source":"ea70788ed894760f.txt","type":"text/plain","size":237},{"uid":"f1817ec99ca7b554","name":"Response Logs","source":"f1817ec99ca7b554.txt","type":"text/plain","size":227}],"parameters":[{"name":"xApiKey","value":"live_utRB76B8Z00RXz4rS0M09nyzvUEGICA7fL0U3GfVzVXxc2mEtU9g8FjwWCrlycT3"},{"name":"imageRequestDto","value":"ImageRequestDto(img=src/main/resources/img/large.jpeg, subId=null, breedIds=Ojos Azules)"},{"name":"statusCode","value":"400"}],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":2,"hasContent":true,"attachmentStep":false}],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":1,"attachmentsCount":2,"hasContent":true,"attachmentStep":false},"afterStages":[],"labels":[{"name":"package","value":"com.thecatapi.api.systemtests.images.uploaded.upload.UploadNewImageNegativeTests"},{"name":"testClass","value":"com.thecatapi.api.systemtests.images.uploaded.upload.UploadNewImageNegativeTests"},{"name":"testMethod","value":"verifyUploadNewImageNegativeTest"},{"name":"parentSuite","value":"Surefire suite"},{"name":"suite","value":"Surefire test"},{"name":"subSuite","value":"com.thecatapi.api.systemtests.images.uploaded.upload.UploadNewImageNegativeTests"},{"name":"host","value":"pkrvmjbmru5nbw0"},{"name":"thread","value":"2199@pkrvmjbmru5nbw0.TestNG-PoolService-2(43)"},{"name":"framework","value":"testng"},{"name":"language","value":"java"},{"name":"epic","value":"Image"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"arg0","value":"Verify that user can not upload image with incorrect breedIds"},{"name":"arg1","value":"ImageRequestDto(img=src/main/resources/img/large.jpeg, subId=null, breedIds=Ojos Azules)"},{"name":"arg2","value":"400"},{"name":"arg3","value":"INVALID_DATA"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Failed tests","traceRegex":".*HttpException.*","matchedStatuses":["broken","failed"],"flaky":false}],"tags":[]},"source":"3e25484dba133222.json","parameterValues":["Verify that user can not upload image with incorrect breedIds","ImageRequestDto(img=src/main/resources/img/large.jpeg, subId=null, breedIds=Ojos Azules)","400","INVALID_DATA"]}